packages/flutter/lib/src/widgets/focus_scope.dart

```diff
@@ -1,145 +1,422 @@
+/// }
+///
+/// @override
+/// Widget build(BuildContext context) {
+///   final TextTheme textTheme = Theme.of(context).textTheme;
+///   return FocusScope(
+///     debugLabel: 'Scope',
+///     autofocus: true,
+///     child: DefaultTextStyle(
+///       style: textTheme.display1,
+///       child: Focus(
+///         onKey: _handleKeyPress,
+///         debugLabel: 'Button',
+///         child: Builder(
+///           builder: (BuildContext context) {
+///             FocusNode focusNode = Focus.of(context);
+///             final bool hasFocus = focusNode.hasFocus;
+///             return RaisedButton(
+///               color: hasFocus ? _color : null,
+///               onPressed: () {
+///                 if (hasFocus) {
+///                   setState((){
+///                     focusNode.unfocus();
+///                   });
+///                 } else {
+///                   setState((){
+///                     focusNode.requestFocus();
+///                   });
+///                 }
+///               },
+///               child: Text(hasFocus ? "I'm in color! Press R,G,B!" : 'Press to focus'),
+///             );
+///           },
+///         ),
+///       ),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
 ///
 /// See also:
 ///
-///  * [FocusScopeNode], which is the associated node in the focus tree.
-///  * [FocusNode], which is a leaf node in the focus tree that can receive
-///    focus.
-class FocusScope extends StatefulWidget {
-  /// Creates a scope in which widgets can receive focus.
+///   * [FocusNode], which represents a node in the focus hierarchy and
+///     [FocusNode]'s API documentation includes a detailed explanation of its
+///     role in the overall focus system.
+///   * [FocusScope], a widget that manages a group of focusable widgets using a
+///     [FocusScopeNode].
+///   * [FocusScopeNode], a node that collects focus nodes into a group for
+///     traversal.
+///   * [FocusManager], a singleton that manages the primary focus and
+///     distributes key events to focused nodes.
+class Focus extends StatefulWidget {
+  /// Creates a widget that manages a [FocusNode].
   ///
-  /// The [node] argument must not be null.
-  const FocusScope({
+  /// The [child] argument is required and must not be null.
+  ///
+  /// The [autofocus] argument must not be null.
+  const Focus({
     Key key,
-    @required this.node,
+    @required this.child,
+    this.focusNode,
     this.autofocus = false,
-    this.child,
-  }) : assert(node != null),
-       assert(autofocus != null),
-       super(key: key);
-
-  /// Controls whether this scope is currently active.
-  final FocusScopeNode node;
+    this.onFocusChange,
+    this.onKey,
+    this.debugLabel,
+  })  : assert(child != null),
+        assert(autofocus != null),
+        super(key: key);
 
-  /// Whether this scope should attempt to become active when first added to
-  /// the tree.
-  final bool autofocus;
+  /// A debug label for this widget.
```

Comment:
Looks like this is unused if an external node is provided? Maybe document that?

Reply:
Documented.